config-controller/internal/controller/policy_controller.go

```diff
@@ -103,44 +102,44 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	// Non deletion case, so either an update or create of a policy
 	desiredState := policyCR.Spec.ToProtobuf()
-	desiredState.Name = policyCR.GetName()
-	desiredState.Source = storage.PolicySource_DECLARATIVE
+	desiredState.Id = policyCR.Status.PolicyId
 
-	existingPolicy, exists, err := r.PolicyClient.GetPolicy(ctx, req.Name)
+	_, exists, err := r.PolicyClient.GetPolicy(ctx, desiredState.GetName())
 	if err != nil {
 		return ctrl.Result{}, errors.Wrap(err, "Failed to fetch policy")
 	}
 
 	var retErr error
-	if exists {
-		desiredState.Id = existingPolicy.Id
-		if err = r.PolicyClient.UpdatePolicy(ctx, desiredState); err != nil {
-			desiredState.Id = existingPolicy.Id
+	if desiredState.Id != "" || exists {
+		if err := r.PolicyClient.UpdatePolicy(ctx, desiredState); err != nil {
 			retErr = errors.Wrap(err, fmt.Sprintf("Failed to update policy %s", req.Name))
 			policyCR.Status.Accepted = false
 			policyCR.Status.Message = retErr.Error()
 		} else {
 			policyCR.Status.Accepted = true
 			policyCR.Status.Message = "Successfully updated policy"
+			policyCR.Status.PolicyId = desiredState.GetId()
 		}
 	} else {
-		if _, err = r.PolicyClient.CreatePolicy(ctx, desiredState); err != nil {
+		if createdPolicy, err := r.PolicyClient.CreatePolicy(ctx, desiredState); err != nil {
 			retErr := errors.Wrap(err, fmt.Sprintf("Failed to create policy %s", req.Name))
 			policyCR.Status.Accepted = false
 			policyCR.Status.Message = retErr.Error()
 		} else {
+			// Create was successful so persist the policy ID received from Central
 			policyCR.Status.Accepted = true
 			policyCR.Status.Message = "Successfully created policy"
+			policyCR.Status.PolicyId = createdPolicy.GetId()
 		}
 	}
 
-	if err = r.K8sClient.Status().Update(ctx, policyCR); err != nil {
+	if err := r.K8sClient.Status().Update(ctx, policyCR); err != nil {
 		return ctrl.Result{}, errors.Wrap(err, fmt.Sprintf("Failed to set status on policy %s", req.Name))
 	}
 
 	if retErr != nil {
 		// Perhaps the cache is stale
-		if err = r.PolicyClient.FlushCache(ctx); err != nil {
+		if err := r.PolicyClient.FlushCache(ctx); err != nil {
```

Comment:
Is this just a best practice?  I don't see any functional difference.

Reply:
The ":" is the difference. Syntax error otherwise.